tests/test_all_rules.py

```diff
@@ -547,6 +547,34 @@ def test_non_bbr_in_correct_dir(self):
                 self.assertEqual(rule.contents.data.building_block_type, None,
                                  f'{self.rule_str(rule)} should be in {proper_directory}')
 
+    def test_rule_change_has_updated_date(self):
+        """Test to ensure modified rules have updated_date field updated."""
+
+        rules_path = get_path("rules")
+        rules__bbr_path = get_path("rules_building_block")
+
+        # Use git diff to check if the file(s) has been modified in rules/ directory
+        # For now this checks even rules/_deprecated but we may combine
+        # with a test case for deprecated rules modification
+        detection_rules_git = make_git()
+        result = detection_rules_git("diff", "--diff-filter=M", "origin/main", "--name-only",
+                                     rules_path, rules__bbr_path)
+
+        # If the output is not empty, then file(s) have changed in the directory(s)
+        if result:
+            rules = result.splitlines()
+            rc = RuleCollection()
+            failed_rules = []
+            for rule_path in rules:
+                rule = rc.load_file(Path(rule_path))
+                last_commit_date = detection_rules_git('log', '-1', '--format=%cd', '--date=short', rule_path)
+                if rule.contents.metadata.updated_date < last_commit_date.replace('-', '/'):
```

Comment:
you can't compare them as strings like this - you need to tuple-ize them first

Reply:
String comparions work fine based on the testing done as well. 
https://pieriantraining.com/python-tutorial-how-to-compare-dates-in-python/